user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/tabs/IFrameTabPanel.java

```diff
@@ -190,83 +190,118 @@ protected void onAttach() {
       attachEventListeners( frame.getElement(), this );
     }
 
-    public native void removeEventListeners( Element ele )
+    // Called when closing a tab. See MantleTabPanel closeTab.
+    public native void removeEventListeners( Element frameElem )
     /*-{
-        var wind = ele.contentWindow;
+        var iframeWindow = frameElem.contentWindow;
+
+        // console.log("IFrameTabPanel.CustomFrame.removeEventListeners url=" + iframeWindow.location.href + " name=" + iframeWindow.name);
+
         try {
-          wind.onmouseup = null;
-          wind.onmousedown = null;
-          wind.onmousemove = null;          
-          wind.onunload = null;
+          iframeWindow.onmouseup = null;
+          iframeWindow.onmousedown = null;
+          iframeWindow.onmousemove = null;
+          iframeWindow.onunload = null;
         } catch(e) {
           // Swallow. Most probably due to Same Domain Origin Policy.
         }
+
         $wnd.watchWindow = null;
     }-*/;
 
-    public native void attachEventListeners( Element ele, CustomFrame frame )
+    public native void attachEventListeners( Element frameElem, CustomFrame frame )
     /*-{
-      var iwind = ele.contentWindow; //IFrame's window instance
-      
-      var funct = function(event){
-        event = iwind.parent.translateInnerMouseEvent(ele, event);
-        iwind.parent.sendMouseEvent(event);
-      }  
-      
-      // Hooks up mouse and unload events
-      $wnd.hookEvents = function(wind){
-        try{
-          if(wind == null){
-            wind = $wnd.watchWindow
-          }
-          wind.onmouseup = funct;
-          wind.onmousedown = funct;
-          wind.onmousemove = funct;
-          
-          wind.onunload = unloader;
-          wind.mantleEventsIn = true;
-          $wnd.watchWindow = null;
-        } catch(e){
-          //You're most likely here because of Cross-site scripting permissions... consuming
-        }
+      // IFrame's window instance.
+      var iframeWindow = frameElem.contentWindow;
+
+      // console.log("IFrameTabPanel.CustomFrame.attachEventListeners url=" + iframeWindow.location.href + " name=" + iframeWindow.name);
```

Comment:
was this left intentionally or debug leftovers?

Reply:
I left this on purpose, to add in any future debugging sessions. Debugging GWT code is particularly hard. Comments are removed if the PRETTY output format is not used, which is the case for production builds.